lib/applications.js

```diff
@@ -56,10 +58,39 @@ apps.Application = Application;
 //   applications - array of JSTP applications
 //
 apps.createAppsIndex = (applications) => {
-  const index = {};
+  const latestApps = new Map();
+  const index = new Map();
 
   applications.forEach((application) => {
-    index[application.name] = application;
+    let appVersions = index.get(application.name);
+    if (!appVersions) {
+      appVersions = new Map();
+      index.set(application.name, appVersions);
+    }
+    if (!application.version) {
+      // no version means default version
+      if (appVersions.get('latest')) {
+        throw new Error(
+          `Multiple entries of '${application.name} without version`
+        );
+      } else {
+        appVersions.set('latest', application);
+      }
+    } else {
+      appVersions.set(application.version.toString(), application);
```

Comment:
Can version be a non-string value?
I thought that we agreed on using the semver, which meant using strings as version values.